frontend/src/app-components/dialogs/DeleteDialog.tsx

```diff
@@ -17,44 +17,79 @@ import {
 } from "@mui/material";
 
 import { DialogTitle } from "@/app-components/dialogs/DialogTitle";
+import { useDelete } from "@/hooks/crud/useDelete";
+import { useDeleteMany } from "@/hooks/crud/useDeleteMany";
 import { DialogControl } from "@/hooks/useDialog";
 import { useTranslate } from "@/hooks/useTranslate";
+import { EntityType } from "@/services/types";
+import { IEntityMapTypes } from "@/types/base.types";
 
-export type DeleteDialogProps<T = string> = DialogControl<T>;
-export const DeleteDialog = <T extends any = string>({
-  open,
-  callback,
-  closeDialog: closeFunction,
-}: DeleteDialogProps<T>) => {
+export type DeleteDialogProps<T extends string = string> = DialogControl<T>;
+export const DeleteDialog = <T extends string = string>({
+  data: ids,
+  entity = EntityType.ATTACHMENT,
+  onDeleteError = () => {},
+  onDeleteSuccess = () => {},
```

Comment:
```suggestion
  onError = () => {},
  onSuccess = () => {},
```
Since it's the delete dialog, it's kindof repetitive to indicate it again.

Reply:
The main motivation of this naming is to avoid overriding the onError attribute of the dialog component